3Dmol/3dmol.js

```diff
@@ -153,19 +153,25 @@ $3Dmol.viewers = {};
  * @param {string} uri - location of data
  * @param {Function} callback - Function to call with arraybuffer as argument.  
  * @param {string} request - type of request
+ * @return {Promise}
  */ 
 $3Dmol.getbin = function(uri, callback, request) {
-    request = (request == undefined)?"GET":request;
-    $.ajax({url:uri, 
-        type: request,
-        dataType: "binary",
-        responseType: "arraybuffer",
-        processData: false}).done(
-            function(ret, txt, response) {
-                callback(ret);
-            }).fail(function(e,txt) { 
-                console.log(txt);
-                });
+    return new Promise(function(resolve, reject) {
+        callback = (callback == undefined)?function(ret){}:callback;
+        request = (request == undefined)?"GET":request;
+        $.ajax({url:uri, 
+            type: request,
+            dataType: "binary",
+            responseType: "arraybuffer",
+            processData: false}).done(
+                function(ret, txt, response) {
+                    callback(ret);
```

Comment:
You can do it this way, but would get more robust error handling if the callback were added to the promise.  Also, creating a dummy functions is a bit inefficient compared to 
```
if(callback) return promise.then(callback);
else return promise;
```